week3.Rmd

```diff
@@ -0,0 +1,209 @@
+
+$$
+\frac{dx}{dt} = \alpha x - \beta xy \\
+\frac{dy}{dt} = \delta xy - \gamma y
+$$
+where
+- x is the prey population (e.g. rabbits)
+- y is the predator population (e.g. wolves)
+- Derivatives $\frac{dy}{dt}$ and $\frac{dx}{dt}$ represent the rate of population change in either population.
+- $\alpha$, $\beta$ $\gamma$ and $\delta$ are constants that describe species-species interactions. 
+
+$\alpha x$ represents prey population growth. This is assumed to be exponential in the presence of adequate nutrition and no predation. This term is counterbalanced by $\beta xy$, which represents loss of prey due to predation.
+
+$\delta xy$ represents the increase in predator population as a result of prey consumption. $\gamma y$ represents loss of predators due to death (or starvation).
+
+In this tutorial, we'll be implementing this model in *Python*, as a launchpad into more complex ecological modeling topics.
+
+In order to store data on animal populations, we will need two arrays, one array each to represent the *predator* and *prey* populations. In each array, the ith element represents the population of each animal at time **i**. For example:
+
+```{python}
+array_1 = [4,5,10,12,20]
+array_2 = [0,2,3,6,8]
+# These arrays are the same length?
+len(array_1) == len(array_2) # True
+```
+
+`array_1` and `array_2` are considered parallel because they 1) have the same dimensions (length = 5) and the values of each index represent two variable instances of the same "observation". Proceeding along these arrays is much like proceeding down a tibble in R.
+
+We can iteratively create (i.e. one step at a time) parallel arrays to store meaningful data! Even more importantly, as we create new data, this new data can be based off pre-existing data in the arrays.
+
+# Example {-}
+
+How can we implement the Lotka-Volterra equations in python using parallel arrays?
+
+First, let's import our friend `matplotlib` as `plt`.
+```{python}
+import matplotlib.pyplot as plt
+```
+
+Because we are working in arrays with discrete indices, we are constrained to think about animal populations in discrete time, which we will call `t` and to which we will assign an arbitrary value (think of it as days).
+
+```{python}
+days = 1000
+```
+
+At `t=0`, the start point of our simulation, we will need to set initial parameters for predator and prey populations. In essence, how many rabbits and wolves were in our forest to start? I'll choose two values arbitrarily.
+
+```{python}
+initPrey = 50
+initPred = 10
+```
+
+Then, we will need to choose our equation constants. Referencing the equations above (and replacing Greek characters with Latin characters), we can set the following rates arbitrarily:
+- Attack rate (a) = 0.2
+- Prey growth rate (r) = 0.01
+- Predator mortality rate (m) = 0.05
+- Conversion constant of prey into predators (k) = 0.1
+
+In real ecological modeling, these sorts of parameters are often based off of empirical data.
+
+```{python}
+a = 0.01
+b = 0.2
```

Comment:
I think there's some confusion here between how you choose values/define symbols used in the markdown section above vs in this code chunk